File: box2dpartone-master/core/src/com/mygdx/game/CollisionListener.java
package com.mygdx.game;

import com.badlogic.gdx.physics.box2d.*;

public class CollisionListener implements ContactListener {
    @Override
    public void beginContact(Contact contact) {
        Fixture fa = contact.getFixtureA();
        Fixture fb = contact.getFixtureB();
        if (fa == null || fb == null) return;
        if (fa.getUserData() == null || fa.getUserData() == null) return;

    }

    @Override
    public void endContact(Contact contact) {

    }

    @Override
    //don't worry about these two
    public void preSolve(Contact contact, Manifold oldManifold) {

    }

    @Override
    public void postSolve(Contact contact, ContactImpulse impulse) {

    }
}

File: src/main/kotlin/io/github/alexlettuce/game/CollisionListener.kt
package io.github.alexlettuce.game

import com.badlogic.gdx.physics.box2d.Contact
import com.badlogic.gdx.physics.box2d.ContactImpulse
import com.badlogic.gdx.physics.box2d.ContactListener
import com.badlogic.gdx.physics.box2d.Manifold

class CollisionListener : ContactListener {
	override fun beginContact(contact: Contact) {
		val fa = contact.fixtureA
		val fb = contact.fixtureB
		if (fa == null || fb == null) return
		if (fa.userData == null || fa.userData == null) return
	}

	override fun endContact(contact: Contact) {}

	//don't worry about these two
	override fun preSolve(contact: Contact, oldManifold: Manifold) {}
	override fun postSolve(contact: Contact, impulse: ContactImpulse) {}
}